conda/common/configuration.py

```diff
@@ -1432,7 +1533,11 @@ def describe_parameter(self, parameter_name):
         if not isiterable(et):
             et = [et]
 
-        if isinstance(parameter._element_type, Parameter):
+        is_class = all(inspect.isclass(element) for element in et)
+
+        # if it's not a Parameter object, we make sure to also exclude all type variables
+        # (i.e. classes) from the next step
+        if isinstance(parameter._element_type, Parameter) or not is_class:
             element_types = tuple(
                 _et.__class__.__name__.lower().replace("parameter", "") for _et in et)
         else:
```

Comment:
Hm, after re-reading this, can we rewrite the whole thing? Isn't the goal to distinguish between classes vs. instances and do different things?

```python
element_types = tuple(
    # if a class is found, return its name as is
    element.__name__
    if inspect.isclass(element)
    # if an instance is found, return a simplified name
    else element.__class__.__name__.lower().replace("parameter", "")
    for element in et
)
```